homework/index.js

```diff
@@ -30,17 +32,134 @@
+    createAndAppend('a', tdForName, {
+      href: repository.html_url,
+      target: '_blank',
+      text: repository.name,
+    });
+    const rowForDescription = createAndAppend('tr', tableElement);
+    createAndAppend('th', rowForDescription, {
+      text: 'Description:',
+    });
+    createAndAppend('td', rowForDescription, { text: repository.description });
+    const rowForForks = createAndAppend('tr', tableElement);
+    createAndAppend('th', rowForForks, {
+      text: 'Forks:',
+    });
+    createAndAppend('td', rowForForks, { text: repository.forks_count });
+    const rowForUpdated = createAndAppend('tr', tableElement);
+    createAndAppend('th', rowForUpdated, {
+      text: 'Updated:',
+    });
+    createAndAppend('td', rowForUpdated, {
+      text: new Date(repository.updated_at).toLocaleString('en-GB'),
+    });
+  }
+
+  function setRepository(repository) {
+    const rightHand = document.getElementById('right-hand');
+    rightHand.innerHTML = '';
+    fetchJSON(repository.contributors_url, (error, users) => {
+      if (error) {
+        renderError(error);
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      const h3Element = createAndAppend('h3', rightHand, {
+        text: 'Contributors:',
+      });
+      if (users.length === 0) {
+        h3Element.textContent = 'Contributor Not Found!';
+        return;
+      }
+      const ulElement = createAndAppend('ul', rightHand, {
+        class: 'contributors',
+      });
+      users.forEach(elem => {
+        const liElement = createAndAppend('li', ulElement, {
+          class: 'contributors',
+        });
+        createAndAppend('img', liElement, {
+          class: 'user-image',
+          src: elem.avatar_url,
+        });
+        createAndAppend('span', liElement, {
+          class: 'login',
+          text: elem.login,
+        });
+        createAndAppend('span', liElement, {
+          class: 'counter',
+          text: elem.contributions,
+        });
+        liElement.onclick = () => {
+          window.open(elem.html_url, '_blank');
+        };
+      });
     });
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-  window.onload = () => main(HYF_REPOS_URL);
+  function getAndAppend(url) {
+    const rootElement = document.getElementById('root');
+    const navElement = createAndAppend('nav', rootElement, {
+      class: 'nav',
+    });
+    createAndAppend('h2', navElement, {
+      class: 'heading-two',
+      text: 'HYF Repositories:',
+    });
+    const divElement = createAndAppend('div', rootElement, {
+      class: 'container',
+      id: 'container',
+    });
+    const select = createAndAppend('select', navElement);
+    fetchJSON(url, (error, repositories) => {
+      if (error) {
+        renderError(error);
+        return;
+      }
+      createAndAppend('option', select, { text: 'Select repository', disabled: 'disabled' });
```

Comment:
The assignment specifies that, at application startup, the details for the first repo in the (sorted) list should be displayed. At present, this happens to be the **alumni** repo. This is in contrast with the other approach we took in the class where we used a placeholder option as you have here, with no details rendered in the page at startup.

Therefore, you should not add this placeholder option at the top of the list. At startup the select box should show the name of the first repo in the list and the page should show the details for that repo.